src/framework/standard/help_commands.rs

```diff
@@ -88,6 +88,46 @@ pub fn has_all_requirements(cmd: &Arc<CommandOptions>, msg: &Message) -> bool {
     !cmd.guild_only
 }
 
+/// Checks whether a command would be visible, takes permissions, channel sent in,
+/// and roles into consideration.
+///
+/// **Note**: A command is visible when it is either normally displayed or
+/// strikethrough upon requested help by a user.
+#[cfg(feature = "cache")]
+pub fn is_command_visible(command_options: &Arc<CommandOptions>, msg: &Message, help_options: &HelpOptions) -> bool {
+    if !command_options.dm_only && !command_options.guild_only
+    || command_options.dm_only && msg.is_private()
+    || command_options.guild_only && !msg.is_private() {
+
+        if let Some(guild) = msg.guild() {
+            let guild = guild.read();
+
+            if let Some(member) = guild.members.get(&msg.author.id) {
+
+                if command_options.help_available {
+
+                    if has_correct_permissions(command_options, msg) {
+
+                        if has_correct_roles(command_options, &guild, &member) {
+                            return true;
+                        } else {
+                            return help_options.lacking_role != HelpBehaviour::Hide;
+                        }
+                    } else {
+                        return help_options.lacking_permissions != HelpBehaviour::Hide;
+                    }
+                } else {
+                    return help_options.lacking_permissions != HelpBehaviour::Hide;
+                }
+            }
+        }
+    } else {
+        return help_options.wrong_channel != HelpBehaviour::Hide;
+    }
+
+    return false;
```

Comment:
At end of function, so this could be changed to an implicit return (just `false`).